### PR TITLE
Fix unexpected GC

### LINKED ISSUE #188: Unexpected GC?
When I build ngx_mruby with MRB_GC_STRESS, some tests failed.
I guess we have some GC related problems. I will look into the details later.

```
diff --git a/build_config.rb b/build_config.rb
index d6cb2f6..22cf6c1 100644
--- a/build_config.rb
+++ b/build_config.rb
@@ -4,6 +4,11 @@ MRuby::Build.new do |conf|

   conf.gembox 'full-core'
+  conf.cc do |cc|
+    cc.defines = %w(GC_PROFILE MRB_GC_STRESS)
+    cc.option_define = '-D%s'
+  end
+
   #
   # Recommended for ngx_mruby
   #
```

```
Fail: ngx_mruby [location /inter_var_file]
 - Assertion[1] Failed: Expected to be equal
    Expected: "fuga => 200 hoge => 400 hoge => 800"
      Actual: "fuga => 200 hoge => 400 hoge => hog"
Fail: ngx_mruby [location /inter_var_inline]
 - Assertion[1] Failed: Expected to be equal
    Expected: "fuga => 100 hoge => 200 hoge => 400"
      Actual: "fuga => 100 hoge => 200 hoge => hog"
Fail: ngx_mruby - upstream keepalive [location /upstream-keepalive]
 - Assertion[1] Failed: Expected to be equal
    Expected: "true"
      Actual: "<html>\r\n<head><title>500 Internal Server Error</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>500 Internal Server Error</h1></center>\r\n<hr><center>nginx/1.9.15</center>\r\n</body>\r\n</html>\r\n"
Fail: ngx_mruby - ssl certificate changing
 - Assertion[1] Failed: Expected to be equal
    Expected: "ssl test ok"
      Actual: ""
 - Assertion[2] Failed: Expected to be equal
    Expected: "1"
      Actual: "0"
Fail: ngx_mruby - ssl certificate changing using data instead of file
 - Assertion[1] Failed: Expected to be equal
    Expected: "ssl test ok"
      Actual: ""
 - Assertion[2] Failed: Expected to be equal
    Expected: "1"
      Actual: "0"
Total: 60
   OK: 55
   KO: 5
Crash: 0
 Time: 1.075196 seconds
```

### PR DESCRIPTION
**Pull request by yyamano**
* Use nginx memory pool to avoid unexpected GC. Fixed #188
* Enable -DMRB_GC_STRESS for test

### CODE REVIEW

**Comment by yyamano** [comment]
Hmm, all builds on travis failed. Will try to test on my dev machine.

File: src/http/ngx_http_mruby_module.c [hunk 1]
```diff
@@ -136,6 +136,11 @@ static ngx_int_t ngx_http_mruby_content_inline_handler(ngx_http_request_t *r);
 static ngx_int_t ngx_http_mruby_log_inline_handler(ngx_http_request_t *r);
 
 #if (NGX_HTTP_SSL) && OPENSSL_VERSION_NUMBER >= 0x1000205fL
+ngx_connection_t *ngx_mruby_connection = NULL;
+ngx_connection_t *ngx_mrb_get_connection(void)
+{
+  return ngx_mruby_connection;
+}
```
Comment (matsumoto-r) [inline src/http/ngx_http_mruby_module.c]: You don't need to use global variable `ngx_connection_t` if `ngx_http_mruby_srv_conf_t *mscf` have `ngx_connection_t ` member like `mscf->c` and get `ngx_http_mruby_srv_conf_t *mscf = mrb->ud; ngx_connection_t *c =  mscf->c` in method functions.

**Comment by matsumoto-r** [comment]
Thank you for your PR. I'll release new version v1.18.3 after merging this PR and #191 !

File: src/http/ngx_http_mruby_module.c (hunk 1 above)
Comment (yyamano) [inline src/http/ngx_http_mruby_module.c]: Fixed in e3d7d2c1ebd8dc9fa12fd3465223a9c958d064f1

**Comment by yyamano** [comment]
#191 is not fixed yet. #188 hits me while working on it :-(

**Comment by yyamano** [comment]
@matsumoto-r I fixed all issues. Could you review it again?

**Comment by matsumoto-r** [comment]
This code looks good to me! Thanks!

**Pull request merged**